client/src/components/Ratings.jsx

```diff
@@ -44,21 +56,53 @@ function Ratings ({product_Id, productName}) {
     };
     fetchMeta();
   }, []);
+
+  useEffect(() => {
+    let filtering = [];
+    for (const star in starsClicked) {
+      if (starsClicked[star] === true) {
+       filtering.push(parseInt(star));
+      }
+    }
+    setStarsFilter(filtering);
+  }, [starsClicked]);
+
+  useEffect(() => {
+    let newList;
+    if (starsFilter.length > 0) {
+      newList = productReviews.results.filter(result =>
+        starsFilter.includes(result.rating)
+      );
+      setReviews(newList);
+    } else {
+      setRevert(!revert);
+    }
+  },[starsFilter]);
+
+  const handleChangeSort = (sortedBy) => {
+    setSort(sortedBy);
+  };
+
   return (
       <div className='reviews'>
         <h3>{`Ratings & Reviews`}</h3>
         <div className='reviews-features'>
           {productReviews.product !== undefined ?
           <ReviewsSidebar
             className='reviews-sidebar'
+            setStarsClicked={setStarsClicked}
+            starsClicked={starsClicked}
             productId={productReviews.product}/> : null}
           <ReviewsList
+            starsClicked={starsClicked}
             className='reviews-list'
-            reviews={productReviews}
+            data={productReviews}
             factors={factors}
             productName={productName}
+            reviews={reviews}
             characteristics={characteristics}
-            product_Id={product_Id}/>
+            product_Id={product_Id}
+            handleChangeSort={handleChangeSort}/>
         </div>
       </div>
   )
```

Comment:
One last comment on this file: in my VSCode, I get an ESLint Parsing Error on Line 26. Do you see this too on your end?

![image](https://user-images.githubusercontent.com/85661360/147868945-d5f3ea16-7127-490f-b454-87e5a18e6300.png)

Reply:
yes, its doing this because we are using a mix of arrow and function declarations. it doesnt like it when you use bothe. can be ignored or if its annoying just comment out the 'func-style' rule in eslintrc.js file